protobuf-provider/src/main/java/io/confluent/kafka/schemaregistry/protobuf/ProtobufSchemaUtils.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ */
+
+package io.confluent.kafka.schemaregistry.protobuf;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.DynamicMessage;
+import com.google.protobuf.Message;
+import com.google.protobuf.util.JsonFormat;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.nio.charset.StandardCharsets;
+
+import io.confluent.kafka.schemaregistry.ParsedSchema;
+
+public class ProtobufSchemaUtils {
+
+  private static final ObjectMapper jsonMapper = new ObjectMapper();
+
+  public static ProtobufSchema copyOf(ProtobufSchema schema) {
+    return new ProtobufSchema(schema.canonicalString(),
+        schema.references(),
+        schema.resolvedReferences(),
+        schema.version()
+    );
+  }
+
+  public static ProtobufSchema getSchema(Object object) {
+    if (object == null) {
+      return null;
+    } else if (object instanceof Message) {
+      Message message = (Message) object;
+      Descriptors.Descriptor desc = message.getDescriptorForType();
+      return new ProtobufSchema(desc);
+    } else {
+      throw new IllegalArgumentException("Unsupported type of class " + object.getClass()
+          .getName());
+    }
+  }
+
+  public static Object toObject(JsonNode value, ParsedSchema parsedSchema) throws IOException {
+    ProtobufSchema schema = (ProtobufSchema) parsedSchema;
```

Comment:
Similar. Since the method is static, can we refactor it to take a ProtobufSchema object instead of ParsedSchema ?

Reply:
Ack